build/src/lib.rs

```diff
@@ -719,18 +707,13 @@ mod tests {
         let builder = BinaryBuilder::new(&["en"], tempdir)
             .version("0.3.0")
             .transform(
-                &|mut buffer, mut writer| {
-                    let mut data = Vec::new();
-                    buffer.read_to_end(&mut data)?;
+                |buffer, mut writer| {
                     let data = smush::encode(
-                        data.as_slice(),
+                        buffer.into_inner().as_slice(),
                         smush::Codec::Zstd,
                         smush::Quality::Maximum,
                     )?;
                     writer.write_all(&data)?;
-
-                    // XXX hangs forever, both with `smush` and `brotli` directly
-                    // let _ = brotli::BrotliCompress(&mut buffer, &mut sink, &Default::default())?;
                     Ok(())
                 },
                 &|p: PathBuf| {
```

Comment:
This ref should go away as well I guess?

Reply:
Good catch!